aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -230,12 +230,25 @@ NestedTensorImpl::NestedTensorImpl(
       nested_sizes_(std::move(nested_sizes)),
       nested_strides_(std::move(nested_strides)),
       storage_offsets_(std::move(storage_offsets)),
-      opt_sizes_(construct_opt_sizes(nested_sizes_)) {
+      opt_sizes_(c10::nullopt) {
```

Comment:
The computation (not allocation) is the expensive part in my opinion. We could allocate this to have the correct size (as determined by the dimension) but then use a binary field to track its content.

Reply:
> The computation (not allocation) is the expensive part in my opinion.

I'm sure this is true, but the current approach makes it very easy to avoid dealing with `opt_sizes` for the symbolic shapes case.